aiida/engine/processes/builder.py

```diff
@@ -242,3 +260,10 @@ def __init__(self, process_class: Type['Process']):
     def process_class(self) -> Type['Process']:
         """Return the process class for which this builder is constructed."""
         return self._process_class
+
+    def _repr_pretty_(self, p, _) -> str:  # pylint: disable=invalid-name
+        """Pretty representation for in the IPython console and notebooks."""
+        return p.text(
+            f'Process class: {self._process_class.__name__}\n'
+            f'Inputs:\n{yaml.safe_dump(json.JSONDecoder().decode(PrettyEncoder().encode(self)))}'
```

Comment:
Doing a full encode/decode cycle in JSON before encoding in YAML is a bit inefficient. Then again, writing serializers in YAML is a bit of a mess (like in `aiida.orm.utils.serialize`, and unfortunately we cannot even reuse that) and this should not be horribly costly, so let's leave this as is.

Reply:
Yeah, I know it's a bit of a messy workaround, but didn't know how to write the serializer in YAML to be honest. I timed the line with `%%timeit` and it only took a couple dozen nanoseconds, so should be ok. ^^